framework/src/media/audio/audio_manager.c

```diff
@@ -140,121 +151,72 @@ static const struct audio_samprate_map_entry_s g_audio_samprate_entry[] = {
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
-audio_manager_result_t find_input_audio_card(void)
+audio_manager_result_t find_audio_card(audio_card_type_t card_type)
 {
 	int ret = AUDIO_MANAGER_SUCCESS;
-	uint8_t card_id = -1;
-	uint16_t device_id = -1;
+	uint8_t card_id = INVALID_ID;
+	uint16_t device_id = INVALID_ID;
 	char type;
 	struct dirent *dir_entry;
 	DIR *dir_info;
 
+	audio_card_info_t *card = g_audio_out_cards;
+	int max_card_num = CONFIG_AUDIO_MAX_OUTPUT_CARD_NUM;
+	char type_chr = 'p';
+	int active_card_id = INVALID_ID;
```

Comment:
I think the role of 'active_card_id' is counting found cards.
So, the initial value can be as zero.
And... how about 'activated_card_count' or 'found_cards' for the variable name?

Reply:
ok I will change.